compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2028,7 +2028,12 @@ where
         TyAndLayout { ty: this.ty, layout }
     }
 
-    fn field(this: TyAndLayout<'tcx>, cx: &C, i: usize) -> C::TyAndLayout {
+    fn ty_and_layout_kind(
+        this: TyAndLayout<'tcx>,
+        cx: &C,
+        i: usize,
+        ty: C::Ty,
+    ) -> Result<C::Ty, C::TyAndLayout> {
```

Comment:
I'd probably add an enum which is 
```rust
// Name can probably we better.
enum TyMaybeWithLayout {
    Ty(C::Ty),
    TyAndLayout(C::TyAndLayout)
}
```

Reply:
Oh I see, I guess that way I don't need to use result and use "Err" as a workaround